src/Control/Category/Tensor.hs

```diff
@@ -10,29 +10,20 @@ import Data.Profunctor
 import Data.These
 import Data.Void
 
-{-
-
-| Tensor | Unit |
-+--------|------+
-| Either | Void |
-|  (,)   |  ()  |
-| These  | Void |
-
-tensor = monoidal structure =
-    category
-  + bifunctor on that category `t`
-  + unit object in that category `i`
-  + isomorphisms in that category `t i a <-> a`, `t a i <-> a`, `t a (t b c) <-> t (t a b) c`
-  + some equalities
-
-monoidal functor = a functor that goes between the underlying
-categories of two different monoidal structure, and has a pair of
-operations `t2 (f a) (f b) -> f (t1 a b)` and `i2 -> f i1`
-
--}
-
-class (Category cat1, Category cat2) => GBifunctor cat1 cat2 r t | t r -> cat1 cat2 where
-  gbimap :: a `cat1` b -> c `cat2` d -> t a c `r` t b d
+-- | A Bifunctor 't' is a binary type operator which allows you to map
+-- over both variables. GBifunctor is the same as the ordinary
+-- 'Data.Bifunctor.Bifunctor' class but we replace the '(->)'s with three different
```

Comment:
Probably worth fleshing out here that we're talking about "arrows" or something. Maybe also some examples for arrows besides `->`.

Reply:
Do you have a good example for an arrow other then `->`?

Reply:
Sure. One example is `<-`, i.e. `Op`.

Reply:
lmao touche